volumes:
  nautilus-test-db-data:

networks:
  nautilus-test-network:

services:
  server:
    build:
      dockerfile: ./docker/Dockerfile.test
      context: .
    ports:
      - "5050:5050"
    volumes:
      - .:/app
    environment:
      - RUST_LOG=nautilus=trace,otel::tracing=trace,otel=debug,tower_http=debug,axum::rejection=trace,diesel_logger=debug
      - DB_URL=postgresql://postgres:postgres@localhost:5433/postgres
      - CARGO_TERM_COLOR= always
      - NAUTILUS_ENVIRONMENT= test
      - AUTH__AUTH_DOMAIN= auth
      - AUTH__AUTH_AUDIENCE= auth
      - AUTH__AUTH_MANAGEMENT_AUDIENCE= auth
      - AUTH__AUTH_MANAGEMENT_CLIENT_ID= auth
      - AUTH__AUTH_MANAGEMENT_SECRET= auth
    depends_on:
      - db
      - migrations
    networks:
      - nautilus-test-network

  migrations:
    build:
      dockerfile: ./docker/Dockerfile.test.migrations
      context: .
    volumes:
      - ./:/app
    depends_on:
      db:
        condition: service_healthy
    restart: "no"
    networks:
      - nautilus-test-network
  db:
    image: postgres:16.0
    restart: always
    attach: false
    volumes:
      - nautilus-test-db-data:/var/lib/postgresql/data
    environment:
      - PGUSER=postgres
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5433:5432"
    healthcheck:
      test: ["CMD", "pg_isready"]
      interval: 5s
      timeout: 5s
      retries: 5
    networks:
      - nautilus-test-network
